server/src/main/java/org/opensearch/common/blobstore/stream/write/WriteContext.java

```diff
@@ -49,7 +51,8 @@ public WriteContext(
         WritePriority writePriority,
         CheckedConsumer<Boolean, IOException> uploadFinalizer,
         boolean doRemoteDataIntegrityCheck,
-        @Nullable Long expectedChecksum
+        @Nullable Long expectedChecksum,
+        Map<String, String> metadata
```

Comment:
Would prefer adding a builder here as we are increasing the args list considerably already.

Reply:
added builder for WriteContext. please take a look.